### PR TITLE
[ADD] wasm build & gitignore

### PR DESCRIPTION
**Pull request by Guerout-Arnaud**
This PR is here to answer and close [this issue](https://github.com/ArthurSonzogni/ftxui-starter/issues/2)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Prerequisites
```
Comment (ArthurSonzogni) [inline .gitignore:1]: The `.gitignore` has never been useful to me and this seems unrelated to your patch. Could you please remove it?

I consider .gitignore a bad practise, because developers are expected to add the files they are interested in. They shouldn't blindly use `git add *`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -31,4 +38,32 @@ target_link_libraries(ftxui-starter
   PRIVATE ftxui::component # Not needed for this example.
 )
 
-install(TARGETS ftxui-starter RUNTIME DESTINATION "bin")
+add_custom_command(TARGET ${PROJECT_NAME}
```
Comment (ArthurSonzogni) [inline CMakeLists.txt:41]: I think you only have to copy the "index.html" and the python script into the build directory.

Could you use:
https://github.com/ArthurSonzogni/termBreaker/blob/6c882381ca41f2cce693e3cb39284d6c281ed5b4/CMakeLists.txt#L216-L220

instead of all those lines? I worry they are going to afraid newcomers ;-)

File: README.md [hunk 3]
```diff
@@ -14,7 +14,19 @@ mkdir build
...
+./ftxui-starter
+~~~
+
+## Webassembly build:
+~~~bash
+mkdir build_emscripten
+cd build_emscripten
+emcmake cmake ..
+make -j3
+cd ../target
```
Comment (ArthurSonzogni) [inline README.md:27]: This line isn't useful anymore, because of my previous comment about not copying files into the "./target" directory.

File: src/main.cpp [hunk 4]
```diff
@@ -1,39 +1,55 @@
...
-  // Limit the size of the document to 80 char.
-  document = document | size(WIDTH, LESS_THAN, 80);
-
-  auto screen = Screen::Create(Dimension::Full(), Dimension::Fit(document));
-  Render(screen, document);
-
-  std::cout << screen.ToString() << std::endl;
-
-  return EXIT_SUCCESS;
+#include <memory>  // for allocator, __shared_ptr_access
```
Comment (ArthurSonzogni) [inline src/main.cpp:1]: This change seems unrelated to your patch. Could you revert it?

**Review by ArthurSonzogni — CHANGES_REQUESTED** [review]
Thank you very much. Here are some comments about the PR:

File: CMakeLists.txt (hunk 2 above)
Comment (Guerout-Arnaud) [reply CMakeLists.txt:41]: I understand your worries, I personally feel that a newcomer would enjoy to have a clear list of files he need to keep and files he can dump (build files) having a target file (or any other name) makes it easier to identify mandatory files for its project to work (at least for wasm part)

waiting for your answer to remove or not this part

File: src/main.cpp (hunk 4 above)
Comment (Guerout-Arnaud) [reply src/main.cpp:1]: I had to include this change since the previous example wasn't working for an unknown reason in wasm. I suppose this is due to the fact that we aren't in a loop and thus it hasn't time to process the render before it exits. 
Maybe this is just an issue on my machine, if you could try it yourself and let me know i would appreciate

**Comment by ArthurSonzogni** [comment]
Thanks for your pull request!
